src/network/webtask.cpp

```diff
@@ -277,6 +305,7 @@ void WebTask::slotAbort() {
             // continuing with the next statements.
             DEBUG_ASSERT(hasTerminated());
             DEBUG_ASSERT(!m_pendingNetworkReplyWeakPtr);
+            doNetworkReplyAborted(pPendingNetworkReply);
```

Comment:
This is not called in our case, because during looping state to there is nothing pending.
The if condition above are false.

Reply:
This is why the timer is always not running.
I think we need another virtual function that is called in the else branch to stop the timer.

Reply:
As far as I tried it was called, I left some qDebug()'s in the MusicBrainzTask::doNetworkReplyAborted and I could see them. But there was nothing we need to do with the pending network reply, so that was quite redundant.

So something like doLoopingTaskAborted(); would work there? Instead of calling doNetworkReplyAborted we could call it.